src/pages/tBTC/Bridge/DepositDetails.tsx

```diff
@@ -54,35 +57,23 @@ export const DepositDetails: PageComponent = () => {
 
   // Extract deposit details values to use them as a dependency in hook
   // dependency array.
-  const btcDepositTxHash = data?.btcTxHash
-  const amount = data?.amount ?? "0"
-  const confirmations = data?.confirmations
-  const requiredConfirmations = data?.requiredConfirmations
-  const optimisticMintingRequestedTxHash =
-    data?.optimisticMintingRequestedTxHash
-  const optimisticMintingFinalizedTxHash =
-    data?.optimisticMintingFinalizedTxHash
-  const thresholdNetworkFee = data?.treasuryFee
-  const mintingFee = data?.optimisticMintFee
+  const {
+    btcTxHash: btcDepositTxHash,
+    amount,
+    confirmations,
+    requiredConfirmations,
+    depositRevealedTxHash,
+    optimisticMintingRequestedTxHash,
+    optimisticMintingFinalizedTxHash,
+    treasuryFee: thresholdNetworkFee,
+    optimisticMintFee: mintingFee,
+  } = data
 
   useEffect(() => {
-    if (
-      !!btcDepositTxHash &&
-      confirmations !== undefined &&
-      requiredConfirmations !== undefined &&
-      confirmations < requiredConfirmations
-    ) {
-      dispatch(
-        tbtcSlice.actions.fetchUtxoConfirmations({
-          utxo: { transactionHash: btcDepositTxHash, value: amount },
-        })
-      )
-    }
-
     return () => {
       updateState("depositDetailsStep", "bitcoin-confirmations")
     }
-  }, [dispatch, btcDepositTxHash, amount, confirmations, requiredConfirmations])
```

Comment:
Besides that, removing this from the hook break the `Minting Timeline`. 

I'm wondering if putting `updateState("depositDetailsStep", "bitcoin-confirmations")` in `return` was a good idea, because it looks it doesn't work as expected tbh 🤔 But still, it would be nice to clear the `depositDetailsStep` somehow when we leave the page.

Reply:
> Besides that, removing this from the hook break the Minting Timeline.

Please elaborate on that, I don't understand what is not working.

> I'm wondering if putting [...] in `return` was a good idea 
> [...] it looks it doesn't work as expected.

That's what `useEffect`'s cleanup function is for. It's called as the element - here `DepositDetails` component is removed from the DOM and it works as expected. The value is set to the first step following the pattern presented in the `mintingStep` value behavior.

Reply:
> Please elaborate on that, I don't understand what is not working.

Yeah, I didn't describe what was wrong, my bad. 

It was not working when the if statement and dependency array was removed (you've already reverted the changes [here](https://github.com/threshold-network/bitcoin-on-base/pull/45#discussion_r1521340041). The reason for that is because without the dependency array, the cleanup function was fired every time the `DepositDetails` component was re-rednered, and it is actually re-rendered a lot of times when some values changes. Because of that, the timeline component always displayed the first step, because the cleanup function was fired when it shouldn't be. Here is a quick video demonstrating that:

https://github.com/threshold-network/bitcoin-on-base/assets/40306834/2464dcd2-b336-4f3d-bf04-8e89c8bd4d56

Both deposit in the viedeo are completed. I had to refresh my memory about useEffect's cleanup function to make figure out why it didn't work properly here. Basically, this is a quick summary of how clean-up function behaves based on dependency array:
1. No Dependency Array: Cleanup runs after every render, before the next effect or on unmount, ensuring effects are properly cleaned up in a constantly updating component.

2. Empty Dependency Array (`[]`): Cleanup runs once, when the component unmounts, mirroring a setup that only needs to run and clean up a single time (like componentDidMount and componentWillUnmount).

3. With Dependencies (`[deps]`): Cleanup runs when any dependency changes, prior to the next effect to handle updated dependencies, and on unmount, allowing for targeted effect management and resource cleanup based on specific state or prop changes.

This means that in our current case, the cleanup function will fire again when (for example) the new confirmation for the bitcoin transaction arrives, thus set the `depositDetailsStep` to `bitcoin-confirmations` . It will work anyway, because we have another `useEffect` below with `confirmations` in dependency array that will set the proper step, but I think I came up with better solution that would be less bug-prone. I've separated the cleanup function to another useEffect with only `depositKey` as a dependency. I've also set it to `undefined` instead of `bitcoin-transactions`.

```
useEffect(() => {
    return () => {
      updateState("depositDetailsStep", undefined)
    }
  }, [depositKey])
```

Result:

https://github.com/threshold-network/bitcoin-on-base/assets/40306834/1f74cb6e-6cf2-4faf-851e-eb9ab6074550

What do you think @kpyszkowski? 

Here is the commit with the fix: c7ec0af3abeb8737823af94acf1ddc248c2658b7

Reply:
Arghh, my bad. I didn't notice I removed the dependency array. Thanks. LGTM